### PR TITLE
✨ Viem migration

### PR DESCRIPTION
**Pull request by jgalat**
Migrates app to use `viem` and ditches `ethers`

### CODE REVIEW

File: .unimportedrc.json [hunk 1]
```diff
@@ -1,4 +1,14 @@
...
+    "sentry.client.config.ts",
+    "sentry.edge.config.ts",
+    "sentry.server.config.ts",
+    "*.config.cjs",
+    "pages/**/*",
+    "scripts/**/*",
+    "i18n/index.ts"
+  ],
+  "ignoreUnused": ["next", "react", "react-dom"],
+  "ignoreUnimported": ["wagmi.config.ts", "cypress.config.ts"]
```
Comment (cruzdanilo) [inline .unimportedrc.json:13]: please, revert

**Review by cruzdanilo — CHANGES_REQUESTED** [review]

File: .unimportedrc.json (hunk 1 above)
Comment (jgalat) [reply .unimportedrc.json:13]: Both `@wagmi/cli` and `cypress` are bringing dependencies which for some reason are conflicting with `unimported`. This is the output I am getting
image

Updating with `npx unimported -u` didn't help. Opted to ignore those 2 entry files, for which I had to narrow `*.config.ts`

Comment (cruzdanilo) [reply .unimportedrc.json:13]: @jgalat what changed about this? every config file is an entry point

Comment (jgalat) [reply .unimportedrc.json:13]: We are now using `@wagmi/cli` for generating the abi types. Its configuration is on `wagmi.config.ts`, which imports stuff from that same package. I suspect these deps are only used in the CLI and that's why they are showing as unimported by the library types. 

Then for `cypress.config.ts`, we had to introduce `@cypress/webpack-preprocessor` (which brings `webpack`, on first line). The issue we had with this is was that e2e tests were failing due to cypress being unable to recognize an optional chaining operator in the new `viem` dependencies. Changing the target from the tsconfig didn't solve this, which lead me to this option. Here is an example of running the tests without it (failing with due to using `_` in numbers)
image

Due to all of this and being unable to just ignore those imports with `unimported -u`, i decided to just remove them from the entry points list

File: .unimportedrc.json [hunk 2]
```diff
@@ -1,4 +1,14 @@
 {
   "ignoreUnresolved": ["i18n"],
-  "entry": ["*.config.ts", "*.config.cjs", "pages/**/*", "scripts/**/*", "i18n/index.ts"]
+  "entry": [
+    "sentry.client.config.ts",
+    "sentry.edge.config.ts",
+    "sentry.server.config.ts",
```
Comment (cruzdanilo) [inline .unimportedrc.json:6]: `sentry.*.config.ts`

File: package.json [hunk 3]
```diff
@@ -5,76 +5,75 @@
...
     "test": "cypress run --config-file cypress.config.ts",
     "ci": "start-server-and-test start:e2e http-get://localhost:3000 test",
     "build": "next build",
-    "build:e2e": "DISABLE_SENTRY_PLUGIN=true NEXT_PUBLIC_NETWORK=mainnet NEXT_PUBLIC_IS_E2E=true next build",
+    "build:e2e": "DISABLE_SENTRY_PLUGIN=true NEXT_PUBLIC_NETWORK=optimism NEXT_PUBLIC_IS_E2E=true next build",
     "start": "NODE_ENV=production next start",
     "start:e2e": "npm run build:e2e && NODE_ENV=production next start",
     "export": "NODE_ENV=production next export",
-    "postinstall": "ts-node -P tsconfig.dev.json scripts/postinstall.ts",
+    "postinstall": "npx wagmi generate",
```
Comment (cruzdanilo) [inline package.json:16]: `wagmi generate`

Comment (jgalat) [reply package.json]: fixed in 01797b8

**Review by cruzdanilo — APPROVED** [review]

**Review by JuampiRombola — APPROVED** [review]

**Pull request merged**